setup/requirements.md

```diff
@@ -43,6 +43,20 @@ currently require the following versions _or newer_:
 If you're using [Remote IDEs](../environments/editors.md), allow pop-ups; Coder
 launches the Remote IDE in a pop-up window.
 
+## Network Policies
+
+Coder uses
+[Container Network Interface (CNI)](https://github.com/containernetworking/cni#what-is-cni)
+plugins to implement network segmentation and tenant isolation in the Kubernetes
+cluster. This enforces network boundaries between pods, preventing users from
+accessing other workspaces.
+
+If there is no CNI plugin configured to enforce such policies, workspaces will
+be able to communicate with each other without restriction.
+
```

Comment:
I'm assuming there's an actionable step here; if so, we could make it clearer to the reader and say, "If you don't configure a CNI plugin to enforce such policies, workspaces will..." and then maybe some info regarding how they might do this (maybe we could tell people that we use Calico and here's how?)

When we say workspaces, we actually mean workspaces (as in environments), right?

Reply:
> When we say workspaces, we actually mean workspaces (as in environments), right?

Correct. And yes, the next actionable step would be to enable a CNI plugin when creating the cluster. Both Azure & GCP have Calico built in, but it requires enabling in the create cluster commands. Enabling Calico in AWS requires some separate commands after the cluster is created.

I'm thinking we say something to the effect that "you'll need to include a CNI plugin compatible with your Kubernetes provider when creating your cluster"

Reply:
We may want to specify a recommendation based on what we test (or state that we test with a specific CNI provider, whatever that is), for the people that are unfamiliar with Kubernetes and don't know which CNI provider to choose.

Reply:
That works with me - I also think we should take @jawnsy's edits into account when saying "Be sure to include..." by making clear that the customer should use a CNI that supports NetworkPolicies.